src/android/java/com/pspdfkit/cordova/action/xfdf/ExportXfdfAction.java

```diff
@@ -0,0 +1,79 @@
+package com.pspdfkit.cordova.action.xfdf;
+
+import android.net.Uri;
+
+import androidx.annotation.NonNull;
+
+import com.pspdfkit.annotations.AnnotationType;
+import com.pspdfkit.cordova.CordovaPdfActivity;
+import com.pspdfkit.cordova.PSPDFKitCordovaPlugin;
+import com.pspdfkit.cordova.action.BasicAction;
+import com.pspdfkit.document.PdfDocument;
+import com.pspdfkit.document.formatters.XfdfFormatter;
+import com.pspdfkit.forms.FormField;
+
+import org.apache.cordova.CallbackContext;
+import org.apache.cordova.PluginResult;
+import org.json.JSONArray;
+import org.json.JSONException;
+
+import java.io.FileNotFoundException;
+import java.io.OutputStream;
+import java.util.Collections;
+import java.util.EnumSet;
+
+import io.reactivex.android.schedulers.AndroidSchedulers;
+import io.reactivex.schedulers.Schedulers;
+
+/**
+ * Asynchronously exports annotations to XFDF-file from the current document.
+ */
+public class ExportXfdfAction extends BasicAction {
+
+  private static final int ARG_XFDF_FILE_URI = 0;
+
+  public ExportXfdfAction(@NonNull String name, @NonNull PSPDFKitCordovaPlugin plugin) {
+    super(name, plugin);
+  }
+
+  @Override
+  protected void execAction(JSONArray args, CallbackContext callbackContext) throws JSONException {
+    final Uri xfdfFileUri = Uri.parse(args.getString(ARG_XFDF_FILE_URI));
+
+    final CordovaPdfActivity cordovaPdfActivity = CordovaPdfActivity.getCurrentActivity();
+    final PdfDocument document = cordovaPdfActivity.getDocument();
+
+    // Capture the given callback and make sure it is retained in JavaScript too.
+    final PluginResult result = new PluginResult(PluginResult.Status.NO_RESULT);
+    result.setKeepCallback(true);
+    callbackContext.sendPluginResult(result);
+
+    if (document != null) {
+      try {
+        final OutputStream outputStream = cordovaPdfActivity.getContentResolver().openOutputStream(xfdfFileUri);
```

Comment:
Is saving to a file also what we do in our iOS wrapper? Isn't this different from the Instant JSON export, which simply returns a string? I wonder if, for consistency, we should also return the XML string here instead of saving to a file, and let the Cordova app handle persistence. WDYT?

Reply:
The API with exporting/importing to/from file is similar to iOS. Because we have plans on merging Cordova and iOS wrapper together we should keep them as close to each other is possible.

@radazzouz Is this correct from your perspective?

Reply:
Yes.